--- rpcbind-0.1.4/src/util.c.orig	2007-10-08 11:45:37.000000000 -0300
+++ rpcbind-0.1.4/src/util.c	2007-10-08 11:46:35.000000000 -0300
@@ -172,7 +172,7 @@ addrmerge(struct netbuf *caller, char *s
 		goto freeit;
 	}
 
-	if (getifaddrs(&ifp) < 0)
+	if (getifaddrs6(&ifp) < 0)
 		goto freeit;
 
 	/*
@@ -346,7 +346,7 @@ network_init()
 	/*
 	 * Now join the RPC ipv6 multicast group on all interfaces.
 	 */
-	if (getifaddrs(&ifp) < 0)
+	if (getifaddrs6(&ifp) < 0)
 		return;
 
 	mreq6.ipv6mr_interface = 0;
