--- ipsec-tools-0.7/src/racoon/crypto_openssl.c.orig	2007-10-19 16:39:22.000000000 -0200
+++ ipsec-tools-0.7/src/racoon/crypto_openssl.c	2007-10-19 16:40:35.000000000 -0200
@@ -41,6 +41,22 @@
 #include <limits.h>
 #include <string.h>
 
+//[GY]05/aug/02 - "defines" included from openssl/sha.h due to conflict when
+//                OpenSSL upgrade to 0.9.8 version
+#define HEADER_SHA_H
+#define SHA_DIGEST_LENGTH 20
+#define SHA_LONG unsigned int
+#define SHA_LBLOCK      16
+
+typedef struct SHAstate_st
+        {
+        SHA_LONG h0,h1,h2,h3,h4;
+        SHA_LONG Nl,Nh;
+        SHA_LONG data[SHA_LBLOCK];
+        unsigned int num;
+        } SHA_CTX;
+//GY
+
 /* get openssl/ssleay version number */
 #include <openssl/opensslv.h>
 
@@ -56,7 +72,9 @@
 #include <openssl/bn.h>
 #include <openssl/dh.h>
 #include <openssl/md5.h>
-#include <openssl/sha.h>
+//[GY]05/aug/02 - excluded due to conflict when OpenSSL upgrade to 0.9.8 version
+//#include <openssl/sha.h>
+//GY
 #include <openssl/hmac.h>
 #include <openssl/des.h>
 #include <openssl/crypto.h>
